crates/scroll/node/src/node.rs

```diff
@@ -0,0 +1,271 @@
+    type Consensus = NoopConsensus;
+
+    async fn build_consensus(self, _ctx: &BuilderContext<Node>) -> eyre::Result<Self::Consensus> {
+        Ok(NoopConsensus::default())
+    }
+}
+
+// #################### NODE STORAGE ####################
+
+/// Storage implementation for Scroll.
+#[derive(Debug, Default, Clone)]
+pub struct ScrollStorage(EthStorage);
+
+impl<Provider> BlockBodyWriter<Provider, BlockBody> for ScrollStorage
+where
+    Provider: DBProvider<Tx: DbTxMut>,
+{
+    fn write_block_bodies(
+        &self,
+        provider: &Provider,
+        bodies: Vec<(u64, Option<BlockBody>)>,
+    ) -> ProviderResult<()> {
+        self.0.write_block_bodies(provider, bodies)
+    }
+
+    fn remove_block_bodies_above(
+        &self,
+        provider: &Provider,
+        block: alloy_primitives::BlockNumber,
+    ) -> ProviderResult<()> {
+        self.0.remove_block_bodies_above(provider, block)
+    }
+}
+
+impl<Provider> BlockBodyReader<Provider> for ScrollStorage
+where
+    Provider: DBProvider + ChainSpecProvider<ChainSpec: EthereumHardforks>,
+{
+    type Block = reth_primitives::Block;
+
+    fn read_block_bodies(
+        &self,
+        provider: &Provider,
+        inputs: Vec<ReadBodyInput<'_, Self::Block>>,
+    ) -> ProviderResult<Vec<BlockBody>> {
+        self.0.read_block_bodies(provider, inputs)
+    }
+}
+
+impl ChainStorage<ScrollPrimitives> for ScrollStorage {
+    fn reader<TX, Types>(
+        &self,
+    ) -> impl reth_provider::ChainStorageReader<
+        reth_provider::DatabaseProvider<TX, Types>,
+        ScrollPrimitives,
+    >
+    where
+        TX: DbTx + 'static,
+        Types: reth_provider::providers::NodeTypesForProvider<Primitives = ScrollPrimitives>,
+    {
+        self
+    }
+
+    fn writer<TX, Types>(
+        &self,
+    ) -> impl reth_provider::ChainStorageWriter<
+        reth_provider::DatabaseProvider<TX, Types>,
+        ScrollPrimitives,
+    >
+    where
+        TX: DbTxMut + DbTx + 'static,
+        Types: NodeTypes<Primitives = ScrollPrimitives>,
+    {
+        self
+    }
+}
+
+// #################### NODE TYPES ####################
+
+impl NodeTypesWithEngine for ScrollNode {
+    type Engine = EthEngineTypes;
+}
+
+impl NodeTypes for ScrollNode {
+    type Primitives = ScrollPrimitives;
+    // TODO (scroll): replace with ScrollChainSpec.
+    type ChainSpec = ChainSpec;
+    // TODO (scroll): replace with BinaryMerklePatriciaTrie.
+    type StateCommitment = MerklePatriciaTrie;
+    type Storage = ScrollStorage;
+}
+
+/// The primitive types for Scroll.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct ScrollPrimitives;
+
+impl NodePrimitives for ScrollPrimitives {
+    type Block = reth_primitives::Block;
+    type BlockHeader = reth_primitives::Header;
+    type BlockBody = BlockBody;
+    type SignedTx = reth_primitives::TransactionSigned;
+    type TxType = reth_primitives::TxType;
+    type Receipt = reth_primitives::Receipt;
```

Comment:
I believe we will need to replace these with the types introduced in #52.

Reply:
For the [`EthereumNode`](https://github.com/paradigmxyz/reth/blob/733f60fe2942ca442d86eb0aba365c6857b92e94/crates/primitives/src/lib.rs#L93-L100), it seems like the `reth_primitives` types are used.

Reply:
Yeah I think once the sdk pattern lands upstream we can migrate to using scroll specific types and remove the requirement for feature flags on the standard types. Would need to look into this further. Lets add a `TODO` for now.